Index: git/os/ipc-os.c
===================================================================
--- git.orig/os/ipc-os.c
+++ git/os/ipc-os.c
@@ -17,7 +17,7 @@
 
 /* Device tree MSCM node compatible property (search key) */
 #if defined(CONFIG_SOC_S32GEN1)
-	#define DT_INTC_NODE_COMP "fsl,s32cc-mscm"
+	#define DT_INTC_NODE_COMP "nxp,s32cc-mscm"
 #elif defined(CONFIG_SOC_S32V234)
 	#define DT_INTC_NODE_COMP "fsl,s32v234-mscm"
 #else
Index: git/os/ipc-uio.c
===================================================================
--- git.orig/os/ipc-uio.c
+++ git/os/ipc-uio.c
@@ -207,7 +207,7 @@ void ipc_os_unmap_intc(void *addr)
 
 static const struct of_device_id ipc_shm_ids[] = {
 	{ .compatible = "fsl,s32v234-mscm", },
-	{ .compatible = "fsl,s32cc-mscm", },
+	{ .compatible = "nxp,s32cc-mscm", },
 	{}
 };
 
