From c877073bc90a8f5b12881b68793fc833caf9a893 Mon Sep 17 00:00:00 2001
From: George Mocanu <george.mocanu@nxp.com>
Date: Fri, 16 Feb 2024 11:32:20 +0200
Subject: [PATCH] arm: Revert GUEST_MAGIC_BASE value

Signed-off-by: George Mocanu <george.mocanu@nxp.com>
---
 xen/include/public/arch-arm.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xen/include/public/arch-arm.h b/xen/include/public/arch-arm.h
index 1cdaf584ee..0eed560ef2 100644
--- a/xen/include/public/arch-arm.h
+++ b/xen/include/public/arch-arm.h
@@ -441,7 +441,7 @@ typedef uint64_t xen_callback_t;
 #define GUEST_GNTTAB_BASE xen_mk_ullong(0x98000000)
 #define GUEST_GNTTAB_SIZE xen_mk_ullong(0x01000000)
 
-#define GUEST_MAGIC_BASE  xen_mk_ullong(0x99000000)
+#define GUEST_MAGIC_BASE  xen_mk_ullong(0x39000000)
 #define GUEST_MAGIC_SIZE  xen_mk_ullong(0x01000000)
 
 #define GUEST_RAM_BANKS   2
-- 
2.25.1

