Index: git/arch/arm64/boot/dts/freescale/s32g.dtsi
===================================================================
--- git.orig/arch/arm64/boot/dts/freescale/s32g.dtsi
+++ git/arch/arm64/boot/dts/freescale/s32g.dtsi
@@ -17,18 +17,26 @@
 
 / {
 	reserved-memory {
-		pfe_reserved_bmu2: pfebufs@34000000 {
+		ipcf_shm: sram@34400000 {
+			compatible = "nxp,s32g-ipcf-shm";
+			/* 1 MB */
+			reg = <0x0 0x34400000 0x0 0x100000>;
+			no-map;
+			status = "okay";
+		};
+
+		pfe_reserved_bmu2: pfebufs@34500000 {
 			compatible = "nxp,s32g-pfe-bmu2-pool";
 			/* BMU2: 512 KB */
-			reg = <0 0x34000000 0 0x80000>;
+			reg = <0 0x34500000 0 0x80000>;
 			no-map;
 			status = "okay";
 		};
 
-		pfe_reserved_rt: pfebufs@34080000 {
+		pfe_reserved_rt: pfebufs@34580000 {
 			compatible = "nxp,s32g-pfe-rt-pool";
 			/* RT: 128 KB */
-			reg = <0 0x34080000 0 0x20000>;
+			reg = <0 0x34580000 0 0x20000>;
 			no-map;
 			status = "okay";
 		};
@@ -50,6 +58,16 @@ pfe_reserved_bdr: pfebufs@835e0000 {
 		/* ends 0x83600000 */
 	};
 
+	mc_rgm: mc_rgm@40078000 {
+		compatible = "fsl,s32gen1-rgm", "nxp,s32g-rgm", "syscon";
+		reg = <0x0 0x40078000 0x0 0x3000>;
+	};
+
+	mc_me: mc_me@40088000 {
+		compatible = "fsl,s32gen1-mc_me", "nxp,s32g-mc_me", "syscon";
+		reg = <0x0 0x40088000 0x0 0x3000>;
+	};
+
 	soc {
 		siul2@4009c000 {
 			compatible = "simple-mfd";
