Index: git/arch/arm64/boot/dts/freescale/s32g-pfe.dtsi
===================================================================
--- git.orig/arch/arm64/boot/dts/freescale/s32g-pfe.dtsi
+++ git/arch/arm64/boot/dts/freescale/s32g-pfe.dtsi
@@ -154,14 +154,12 @@ pfe_netif2: ethernet@12 {
 			/* Network interface 'aux0' */
 			pfe_aux0: ethernet@13 {
 				compatible = "nxp,s32g-pfe-netif", "fsl,pfeng-logif";
-				status = "disabled";
+				status = "okay";
 				reg = <13>;
 				local-mac-address = [ 00 04 9F BE EF 80 ];
 				nxp,pfeng-if-name = "aux0";
 				fsl,pfeng-if-name = "aux0";
-				nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_0>,
-							 <PFE_HIF_CHANNEL_1>,
-							 <PFE_HIF_CHANNEL_2>;
+				nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_2>;
 				nxp,pfeng-netif-mode-aux;
 			};
 
