Index: git/arch/arm64/boot/dts/freescale/fsl-s32g.dtsi
===================================================================
--- git.orig/arch/arm64/boot/dts/freescale/fsl-s32g.dtsi
+++ git/arch/arm64/boot/dts/freescale/fsl-s32g.dtsi
@@ -152,7 +152,7 @@ pfe_hif2: hif@2 {
 			status = "okay";
 			interrupts = <GIC_SPI 192 IRQ_TYPE_EDGE_RISING>;
 			reg = <2>;
-			fsl,pfeng-hif-mode = <PFENG_HIF_MODE_EXCLUSIVE>;
+			fsl,pfeng-hif-mode = <PFENG_HIF_MODE_SHARED>;
 		};
 
 		/* PFE_HIF_3 */
@@ -300,11 +300,11 @@ pfe_aux0: ethernet@3 {
 			compatible = "fsl,pfeng-logif";
 			#address-cells = <1>;
 			#size-cells = <0>;
-			status = "disabled";
+			status = "okay";
 			local-mac-address = [ 00 04 9F BE EF 80 ];
 			fsl,pfeng-if-name = "aux0";
 			fsl,pfeng-logif-mode = <PFENG_LOGIF_MODE_AUX>;
-			fsl,pfeng-hif-channels = <&pfe_hif1>;
+			fsl,pfeng-hif-channels = <&pfe_hif2>;
 		};
 	};
 
