From 61dfd00465171e2c41832d92da96e966bbea6e44 Mon Sep 17 00:00:00 2001
From: Mihai Cacior <mihai.cacior@nxp.com>
Date: Tue, 18 Jun 2024 16:18:50 +0300
Subject: [PATCH] sja1110: add power management resume handler

Signed-off-by: Mihai Cacior <mihai.cacior@nxp.com>
---
 sja1110_init.c | 57 ++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 57 insertions(+)

diff --git a/sja1110_init.c b/sja1110_init.c
index b8d4eaf..5d85f6d 100644
--- a/sja1110_init.c
+++ b/sja1110_init.c
@@ -1111,6 +1111,62 @@ static const struct of_device_id sja1110_dt_ids[] = {
 };
 MODULE_DEVICE_TABLE(of, sja1110_dt_ids);
 
+#ifdef CONFIG_PM_SLEEP
+static int sja1110_resume(struct device *device)
+{
+	// Wait 0.1 seconds before probing the sja1110.
+	usleep_range(100000, 100001);
+
+	int ret = 0;
+	struct spi_device *spi = container_of(device, struct spi_device, dev);
+	struct sja1110_priv *sja1110 = spi_get_drvdata(spi);
+
+	INIT_WORK(&sja1110->work, sja1110_init_hw_worker);
+
+	/* device specific probe */
+	if (sja1110->devtype == SJA1110_UC) {
+		/* µC firmware */
+		dev_info(&spi->dev, "probing uc\n");
+
+		/* indicate that µC data structures are initialized */
+		g_state.sja1110[sja1110->devtype] = sja1110;
+		complete(&g_state.uc_initialized);
+	} else {
+		/* switch static config */
+		dev_info(&spi->dev, "probing switch\n");
+
+		/* GPIO driver init */
+		ret = sja1110_gpio_setup(sja1110);
+		if (ret) {
+			dev_err(&spi->dev,
+				"Could not register gpiochip (ret=%d)\n", ret);
+			return ret;
+		}
+
+		/* try to get a GPIO pin from the device tree */
+		sja1110->switch_priv->rst_gpio = get_and_request_gpio(sja1110);
+
+		/**
+		 * Try to upload binaries with default names after probing.
+		 * Upload switch config FIRST, µC firmware SECOND.
+		 * Wait until µC is probed BEFORE starting the fw upload
+		 * (switch is already probed when we reach this section).
+		 * Synchronization via 'uc_initialized' completion.
+		 */
+		g_state.sja1110[sja1110->devtype] = sja1110;
+		schedule_work(&g_state.post_probe_work);
+	}
+
+	return ret;
+}
+
+static SIMPLE_DEV_PM_OPS(sja1110_pm_ops, NULL, sja1110_resume);
+
+#define SJA1110_PM_OPS (&sja1110_pm_ops)
+#else
+#define SJA1110_PM_OPS (NULL)
+#endif /* CONFIG_PM_SLEEP */
+
 static struct spi_driver sja1110_driver = {
 	.probe  = sja1110_probe,
 	.remove = sja1110_remove,
@@ -1118,6 +1174,7 @@ static struct spi_driver sja1110_driver = {
 		.name = "sja1110",
 		.owner = THIS_MODULE,
 		.of_match_table = of_match_ptr(sja1110_dt_ids),
+		.pm = SJA1110_PM_OPS,
 	},
 };
 
-- 
2.39.1.windows.1
