diff --git a/plat/nxp/s32/include/s32_platform_def.h b/plat/nxp/s32/include/s32_platform_def.h
index cb3b32457..6c5639f87 100644
--- a/plat/nxp/s32/include/s32_platform_def.h
+++ b/plat/nxp/s32/include/s32_platform_def.h
@@ -100,8 +100,7 @@
  * enough to prevent overflowing onto the adjacent SRAM image. Handle with care,
  * wear a helmet and compile with -Os.
  */
-#define BOOTROM_ADMA_RSRVD_BASE		(0x343ff000)
-#define BL2_LIMIT					(BOOTROM_ADMA_RSRVD_BASE - 1)
+#define BL2_LIMIT               (0x34700000 - 1)
 
 /* U-boot addresses in DDR.
  * BL33_MAX_DTB_SIZE and BL33_ENTRYPOINT must be kept in sync
diff --git a/plat/nxp/s32/s32_common.mk b/plat/nxp/s32/s32_common.mk
index e917f1bda..ec9deb162 100644
--- a/plat/nxp/s32/s32_common.mk
+++ b/plat/nxp/s32/s32_common.mk
@@ -444,7 +444,7 @@ endif
 ifneq (${HSE_SUPPORT},)
 BL2_BASE		?= 0x34082000
 else
-BL2_BASE		?= 0x34302000
+BL2_BASE		?= 0x34610000
 endif
 $(eval $(call add_define,BL2_BASE))
 
