Index: git/plat/nxp/s32/s32_common.mk
===================================================================
--- git.orig/plat/nxp/s32/s32_common.mk
+++ git/plat/nxp/s32/s32_common.mk
@@ -388,7 +388,7 @@ ${MKIMAGE_FIP_CONF_FILE}: ${BL2_W_DTB_SIZE_FILE} ${FIP_HDR_SIZE_FILE} add_to_fip
 	T_SIZE=0x$$($(call hexbc, $${BL2_W_DTB_SIZE}, +, $${HDR_SIZE})); \
 	echo "DATA_FILE SIZE $$T_SIZE" >> $@
 
-FIP_ALIGN := 16
+FIP_ALIGN := 64
 all: add_to_fip
 add_to_fip: fip ${BL2_W_DTB}
 	$(eval FIP_MAXIMUM_SIZE_10 = $(shell printf "%d\n" ${FIP_MAXIMUM_SIZE}))
