Index: 1.0-r7/interfaces
===================================================================
--- 1.0-r7.orig/interfaces
+++ 1.0-r7/interfaces
@@ -9,19 +9,19 @@ auto pfe0
 iface pfe0 inet manual
 pre-up [ -L /sys/class/net/pfe0 ]
 dhcpcd_opts -LGb
-up /bin/ifconfig pfe0 up
+up /sbin/ifconfig pfe0 up
 
 auto pfe1
 iface pfe1 inet manual
 pre-up [ -L /sys/class/net/pfe1 ]
 dhcpcd_opts -LGb
-up /bin/ifconfig pfe1 up
+up /sbin/ifconfig pfe1 up
 
 auto pfe2
 iface pfe2 inet manual
 pre-up [ -L /sys/class/net/pfe2 ]
 dhcpcd_opts -LGb
-up /bin/ifconfig pfe2 up
+up /sbin/ifconfig pfe2 up
 
 auto eth0
 iface eth0 inet dhcp
