Index: s32g274ardb2/interfaces
===================================================================
--- s32g274ardb2.orig/interfaces
+++ s32g274ardb2/interfaces
@@ -7,6 +7,7 @@ iface lo inet loopback
 auto pfe0
 iface pfe0 inet dhcp
 pre-up [ -L /sys/class/net/pfe0 ]
+pre-up ethtool -K pfe0 tx off
 dhcpcd_opts -Gb
 
 auto pfe1
@@ -17,6 +18,7 @@ dhcpcd_opts -Gb
 auto pfe2
 iface pfe2 inet dhcp
 pre-up [ -L /sys/class/net/pfe2 ]
+pre-up ethtool -K pfe2 tx off
 dhcpcd_opts -Gb
 
 auto eth0
