Index: 1.0-r7/interfaces
===================================================================
--- 1.0-r7.orig/interfaces
+++ 1.0-r7/interfaces
@@ -30,6 +30,37 @@ dhcpcd_opts -LGb
 up /sbin/ifconfig aux0 up
 
 auto eth0
+iface eth0 inet manual
+dhcpcd_opts -LGb
+up /sbin/ifconfig eth0 up
+pre-up [ -L /sys/class/net/eth0 ]
 iface eth0 inet dhcp
 pre-up [ -L /sys/class/net/eth0 ]
-dhcpcd_opts -LGb
+pre-up [ ! -L /sys/class/net/can0 ]
+dhcpcd_opts -Lb
+
+auto xenbr0
+iface xenbr0 inet dhcp
+pre-up [ -L /sys/class/net/can0 ]
+pre-up brctl addbr xenbr0
+pre-up echo 0 > /sys/class/net/xenbr0/bridge/default_pvid
+pre-up brctl addif xenbr0 eth0
+pre-up ip addr flush dev eth0
+pre-up ip link set dev eth0 up
+post-down brctl delif xenbr0 eth0
+post-down brctl delbr xenbr0
+dhcpcd_opts -Lb
+
+auto eth1
+iface eth1 inet static
+pre-up [ ! -L /sys/class/net/can0 ]
+address 10.0.100.20
+netmask 255.255.255.0
+
+auto v2xbr
+iface v2xbr inet static
+pre-up [ -L /sys/class/net/can0 ]
+pre-up brctl addbr v2xbr
+post-down brctl delbr v2xbr
+address 10.0.100.10
+netmask 255.255.255.0
